src/main/scala/calc/Compiler.scala

```diff
@@ -66,17 +73,129 @@ object Compiler {
    *
    *  This is the main method you have to implement.
    */
-  def compileExpr(tree: Tree): irt.Tree = {
+  def compileExpr(tree: Tree, scope: Scope=Map.empty[String, TreeType]): irt.Tree = {
     // TODO
     implicit val pos = tree.pos
 
     tree match {
-      case Literal(value) =>
+      case Literal(value, _) =>
         irt.DoubleLiteral(value)
 
+      case i@Ident(name, tp) => scope.get(name) match {
+        // we have to cast parameters to Double, as they are always of type AnyType:
+        case Some(t) if tp.isParam =>
+          irt.Unbox(irt.VarRef(irt.Ident(name))(irtpe.AnyType), 'D')
```

Comment:
This works, but is not the most efficient way to deal with the params-are-any problem. A better way is to do what is typically done by Java's and Scala's erasure: give the parameters mangled/generated names, and extract their double value in local variables of type `Double` at the beginning of the body of the closure.